docs/src/pages/demos/breadcrumbs/RouterBreadcrumbs.tsx

```diff
@@ -0,0 +1,148 @@
+  '/trash': 'Trash',
+  '/spam': 'Spam',
+  '/drafts': 'Drafts',
+};
+
+function ListItemLink(props: ListItemLinkProps) {
+  const { to, open, ...other } = props;
+  const primary = breadcrumbNameMap[to];
+
+  return (
+    <li>
+      <ListItem button component={RouterLink} to={to} {...other}>
+        <ListItemText primary={primary} />
+        {open != null ? open ? <ExpandLess /> : <ExpandMore /> : null}
+      </ListItem>
+    </li>
+  );
+}
+
+ListItemLink.propTypes = {
+  open: PropTypes.bool,
+  to: PropTypes.string.isRequired,
+};
+
+const styles = (theme: Theme) =>
+  createStyles({
+    root: {
+      display: 'flex',
+      flexDirection: 'column',
+      width: 360,
+    },
+    lists: {
+      backgroundColor: theme.palette.background.paper,
+      marginTop: theme.spacing(1),
+    },
+    nested: {
+      paddingLeft: theme.spacing(4),
+    },
+  });
+
+interface LinkRouterProps extends LinkProps {
+  to: string;
+  replace?: boolean;
+}
+
+const LinkRouter = (props: LinkRouterProps) => <Link {...props} component={RouterLink as any} />;
+
+class RouterBreadcrumbs extends React.Component<RouterBreadcrumbsProp, RouterBreadcrumbsState> {
+  state = {
+    open: true,
+  };
+
+  handleClick = () => {
+    this.setState(state => ({ open: !state.open }));
+  };
+  static propTypes: any;
+
+  render() {
+    const { classes } = this.props;
+
+    // Use NoSsr to avoid SEO issues with the documentation website.
+    return (
+      <NoSsr>
+        <MemoryRouter initialEntries={['/inbox']} initialIndex={0}>
+          <div className={classes.root}>
+            <Route>
+              {({ location }) => {
+                const pathnames = location.pathname.split('/').filter(x => x);
+
+                return (
+                  <Breadcrumbs aria-label="Breadcrumb">
+                    <LinkRouter color="inherit" to="/">
+                      Home
+                    </LinkRouter>
+                    {pathnames.map((value, index) => {
+                      const last = index === pathnames.length - 1;
+                      const to = `/${pathnames.slice(0, index + 1).join('/')}`;
+
+                      return last ? (
+                        <Typography color="textPrimary" key={to}>
+                          {breadcrumbNameMap[to]}
+                        </Typography>
+                      ) : (
+                        <LinkRouter color="inherit" to={to} key={to}>
+                          {breadcrumbNameMap[to]}
+                        </LinkRouter>
+                      );
+                    })}
+                  </Breadcrumbs>
+                );
+              }}
+            </Route>
+            <div className={classes.lists}>
+              <List component="nav">
+                <ListItemLink to="/inbox" open={this.state.open} onClick={this.handleClick} />
+                <Collapse in={this.state.open} timeout="auto" unmountOnExit>
+                  <List component={'div' as 'ul'} disablePadding>
```

Comment:
cc @eps1lon

Reply:
It's not ideal but until we make the ListProps generic this is the workaround I've recommended in the past.

Reply:
Resolved with #15292